pyrules/rules.py

```diff
@@ -1,93 +1,161 @@
-        if translations:
-            translator = Translator(translations)
-            for rule in self.rules:
-                for key in rule.keys():
-                    rule[key] = [translator.replace(item) for item in rule[key]]
-    def should_trigger(self, context):
-        return True
+        if name:
+            self.name = name
+        self._current_ruleid = None
+
     def perform(self, context):
         count = 0
         for rule in self.rules:
-            if all([eval(condition, context.as_dict) for condition in rule['if']]):
+            evaluator = LogicEvaluator(
+                rule['if'].get('logic'), rule['if']['conditions'])
+            if evaluator.evaluate(context):
                 count = count + 1
-                self._current_ruleid = rule.get('id', count)
+                self._current_ruleid = rule.get('rule') or count
                 for action, target in zip(rule['then'], rule['target']):
-                    if context._translations:
-                        action = context._translations.replace(action)
-                        target = context._translations.replace(target)
-                    result = context[target.replace('context.', '').strip()] = eval(action, context.as_dict)
+                    #if context._translations:
+                    #    action = context._translations.replace(action)
+                    #    target = context._translations.replace(target)
+                    result = \
+                        context[target.replace('context.', '').strip()] = (
+                            eval(action, context.as_dict)
+                            if isinstance(action, basestring)
+                            else action)
                     self.record(context, result)
             else:
-                break
+                continue
         else:
             self._current_ruleid = None
             return True
         return False
+
     @property
     def ruleid(self):
         if self._current_ruleid:
-            return "%s.%s" % (super(TableRuleset, self).ruleid, self._current_ruleid)
-        return super(TableRuleset, self).ruleid
-    
+            return "%s.%s" % (super(TableRule, self).ruleid, self._current_ruleid)
+        return super(TableRule, self).ruleid
+
+    @classmethod
+    def from_yaml(cls, text):
+        return cls._from_data(yaml.load(text))
+        
+    @classmethod
+    def from_json(cls, text):
+        return cls._from_data(json.loads(text))
+
+    @classmethod
+    def _from_data(cls, data):
+        # We have to convert non-string data in clauses back to strings,
+        # because they will be eval-ed
+        rules = []
+        for rule in data['rules']:
+            obj = {
+                'rule': rule.get('rule'),
+                'then': rule['then'],
+                'target': rule['target']}
+            if_clause = {}
+            # Convert conditions to dictionaries, i.e. "foo" becomes {"foo": True}
```

Comment:
Yeah, I understand that it's possible to use translations here. What I don't get is how this would be useful in real world.

In case of NaturalLanguageRule it makes sense, since it's effectively a DSL for rule specification and we naturally want to be able to localize it. But if we have a machine-readable format like YAML/JSON, why would we have foreign languages inside structured data?

I mean, if it's generated by some tool, it should be outputting correct variable names already and handling i18n in it's UI. And if it's written manually by human, then specifying correct variable names is way simpler than understanding how to write YAML :-) . It just seems like translation doesn't belong to TableRule at all.